core/controllers/contributor_dashboard.py

```diff
@@ -276,12 +358,16 @@ def _get_reviewable_exploration_opportunity_summaries(
             reference.story_id
             for topic in topics
             for reference in topic.get_all_story_references()
-            if reference.story_is_published])
-        topic_exp_ids = [
-            node.exploration_id
-            for story in topic_stories
-            for node in story.story_contents.get_ordered_nodes()
-        ]
+            if reference.story_is_published], strict=True)
+        topic_exp_ids = []
+        for story in topic_stories:
+            for node in story.story_contents.get_ordered_nodes():
+                # Here 'node.exploration_id' can never be None because
+                # 'node.exploration_id' can only be None initially, when
+                # the story creator has just created a story with the
+                # basic storyline.
+                assert node.exploration_id is not None
```

Comment:
I don't think I follow the explanation for why an assert is sensible here -- this seems based on general reasoning and not a condition from earlier in the code. So, if we don't expect node.exploration_id to be None in practice, then shouldn't we raise an exception instead?

Reply:
> shouldn't we raise an exception instead?

Done